seacatauth/openidconnect/handler/introspect.py

```diff
@@ -120,7 +120,8 @@ async def introspect(self, request):
 		if "preferred_username" in user_info:
 			response_data["username"] = user_info["preferred_username"]
 
-		# TODO: Verify that the token is  client_id is included in the aud claim
+		# TODO: Verify that the requesting client is part of the token's intended audience
+		#  (i.e. that their client_id is included in the aud claim).
```

Comment:
This will break my usecase in mobile apps. Can we have a call about thiz plz?

Reply:
i don't intend to implement this check within this merge request (or anytime soon, unless necessary). it would also require implementing a way for the client to influence the content of the `audience` claim somehow, possibly by `scope`.

i added the todo as a reminder that the authorization has some limitations and can be improved. the authorization server should not disclose id tokens to just any client. it is mostly fine in environments where there are few clients manually registered by an admin, but not in large environments with open client registration.

Reply:
Clarified on call